feat: Create api routes

nahharris opened the pull request:
This PR creates the 3 needed API routes, but don't implement them.

nahharris committed:
feat: Create API routes

Create the API routes for:

- `/especialidade`
- `/unidades?cep=&eps=`
- `/detalhes?cnes=`

The routes aren't functional yet, just implemented

nahharris committed:
docs: Add basic documentation for each route

Included basic documentation for each route, visible at `/docs` swagger
page

Signed-off-by: OJarrisonn <j.h.m.t.v.10@gmail.com>

Marcoszz commented on src/app.py:
```diff
@@ -5,4 +5,16 @@
+async def unidades(cep: str, esp: str):
+    return {"message": f"Não implementado ainda! Unidades próximas ao CEP {cep} com especialidade {esp}"}
+
+@app.get("/detalhes")
```
Troca o nome dessa rota para `/unidades/detalhes`

nahharris replied:
Ok, embora no projeto que tinhamos feito originalmente a rota se chamava apenas `/detalhes`

Marcoszz commented on src/app.py:
```diff
@@ -5,4 +5,16 @@
 
 @app.get("/")
 async def root():
```
Pode excluir essa rota

nahharris replied:
Resolvi manter para ser uma rota para você checar rapidamente se o servidor está rodando digitando apenas `localhost:8000` no navegador

Marcoszz replied:
Tudo bem!

Marcoszz reviewed (changes requested):
![image](https://github.com/user-attachments/assets/deb8bdee-e1da-44a0-9162-dad8c9721a52)
Tô tomando _Internal Server Error_ em todas as rotas, pode verificar?

Marcoszz commented:
Also, vcs precisam que a aplicação rode com Python 3.12? Estou usando o 3.10 aqui. Se sim, eu aconselho fortemente vcs a usarem Docker.

Outra sugestão, só que importante: seria interessante vocês criarem algum `Makefile` ou `.sh` pra fazer todo esse processo de preparação da `.venv` e start da aplicação. Eu segui o tutorial diretinho e tá de boas, mas seria algo legal pra vocês mesmos!

nahharris committed:
fix: routes now attend to type checking

The response for the routes `/especialidades` and `/unidades` should
return a `list[str]` and a `list[dict]` but were returning a `dict`

Signed-off-by: OJarrisonn <j.h.m.t.v.10@gmail.com>

nahharris committed:
refactor: Rename `/detalhes` to `/unidades/detalhes`

Renames the route `/detalhes` to `/unidades/detalhes` as asked by
@Marcoszz

Signed-off-by: OJarrisonn <j.h.m.t.v.10@gmail.com>

Marcoszz commented on src/app.py:
```diff
@@ -1,20 +1,45 @@
+
+- `/especialidades`: Retorna a lista de especialidades disponíveis.
+- `/unidades?cep=&esp=`: Retorna a lista de unidades de saúde próximas a um CEP com uma determinada especialidade.
+- `/detalhes?cnes=`: Retorna detalhes de uma unidade de saúde a partir do numero CNES da unidade.
```
Trocar o nome da rota de detalhes na descrição

nahharris committed:
docs: Update `/detalhes` to `/unidades/detalhes` in app description

Signed-off-by: OJarrisonn <j.h.m.t.v.10@gmail.com>
